cis194/week4/robert-chang/src/Lib.hs

```diff
@@ -0,0 +1,29 @@
+module Lib where
+
+--Task1
+
+fun1 :: [Integer] -> Integer
+fun1 [] = 1
+fun1 (x:xs)
+ | even x = (x - 2) * fun1 xs
+ | otherwise = fun1 xs
+
+fun1' :: [Integer] -> Integer
+--fun1' = product . fmap (subtract 2) . filter even
+--fun1' = product . map (\x -> x - 2) . filter even
+fun1' = product . (subtract 2 <$>) . filter even
+
+
+fun2 :: Integer -> Integer
+fun2 1 = 0
+fun2 n | even n = n + fun2 (n `div` 2)
+ | otherwise = fun2 (3 * n + 1)
+
+fun2' :: Integer -> Integer
+fun2' = sum 
+ . filter even
+ . takeWhile (>1) 
+ . iterate (\n -> 
+    if even n
+      then n `div` 2
+      else 3 * n + 1) 
```

Comment:
we normally align if, then and else like this:
```Haskell
if even n
then n `div` 2
else 3 * n + 1
```

Reply:
@shineli1984 thanks!

Also uploaded another solution for this task. Plz check.